restful.module

```diff
@@ -749,3 +768,21 @@ function restful_date_time_format_element_validate($element, &$form_state) {
     )));
   }
 }
+
+/**
+ * Determine if this is the first time we try to switch the user.
+ *
+ * @return bool
+ *   TRUE if this static function was already called.
+ */
+function restful_is_user_switched() {
```

Comment:
I think that it's OK. You are not supposed to create multiple `AuthenticationManager`. If you do, you need to be aware of the insights of doing so.